parachain/pallets/omni-account/src/lib.rs

```diff
@@ -0,0 +1,218 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+pub use core_primitives::{Identity, MemberIdentity};
+pub use frame_system::pallet_prelude::BlockNumberFor;
+pub use pallet::*;
+
+use frame_support::pallet_prelude::*;
+use frame_system::pallet_prelude::*;
+use sp_core::H256;
+use sp_core_hashing::blake2_256;
+use sp_std::vec::Vec;
+
+#[derive(Encode, Decode, TypeInfo, Clone, PartialEq, Eq, RuntimeDebug)]
+pub struct MemberAccount {
+	pub id: MemberIdentity,
+	pub hash: H256,
+}
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+
+	/// The current storage version.
+	const STORAGE_VERSION: StorageVersion = StorageVersion::new(0);
+
+	#[pallet::pallet]
+	#[pallet::storage_version(STORAGE_VERSION)]
+	#[pallet::without_storage_info]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// The event type of this pallet.
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+		/// The origin which can manage the pallet.
+		type TEECallOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+		/// The maximum number of identities an id graph can have.
+		#[pallet::constant]
+		type MaxIDGraphLength: Get<u32>;
+	}
+	pub type IDGraphMembers<T> = BoundedVec<MemberAccount, <T as Config>::MaxIDGraphLength>;
+
+	#[pallet::storage]
+	pub type LinkedIdentityHashes<T: Config> =
+		StorageMap<Hasher = Blake2_128Concat, Key = H256, Value = ()>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn id_graphs)]
+	pub type IDGraphs<T: Config> =
+		StorageMap<Hasher = Blake2_128Concat, Key = Identity, Value = IDGraphMembers<T>>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn id_graph_hashes)]
+	pub type IDGraphHashes<T: Config> =
+		StorageMap<Hasher = Blake2_128Concat, Key = Identity, Value = H256>;
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// Identity linked
+		IdentityLinked { who: Identity, identity: H256 },
+		/// Identity remove
+		IdentityRemoved { who: Identity, identity_hashes: Vec<H256> },
+		/// Identity made public
+		IdentityMadePublic { who: Identity, identity_hash: H256 },
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Identity is already linked
+		IdentityAlreadyLinked,
+		/// IDGraph len limit reached
+		IDGraphLenLimitReached,
+		/// Identity not found
+		IdentityNotFound,
+		/// Invalid identity
+		PrimeIdentityInvalid,
+		/// Prime identity not found
+		PrimeIdentityNotFound,
```

Comment:
Is the concept of prime identity still valid, mostly it refers to `who` ?
Only this and variant above mentions it.

Maybe it could be changed to:
`PrimeIdentityInvalid` -> `InvalidIdentity` 
`PrimeIdentityNotFound` -> `UnknownIdGraph`